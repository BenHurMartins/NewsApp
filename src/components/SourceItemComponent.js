import React from 'react';
import {Text, StyleSheet, TouchableOpacity, View, Linking} from 'react-native';
import {Colors, Mixins} from '../styles';
import {Divider, UrlTextComponent} from './';
import {useDispatch} from 'react-redux';
import {SET_ID, SET_SOURCE} from '../reducers/actionsTypes';
import {useDeviceOrientation} from '@react-native-community/hooks';
import {useNavigation} from '@react-navigation/native';

const SourceItemComponent = ({source}) => {
  const navigation = useNavigation();
  const dispatch = useDispatch();
  const orientation = useDeviceOrientation();

  const handleClick = () => {
    dispatch({type: SET_ID, payload: source.id});
    dispatch({type: SET_SOURCE, payload: source.name});
    if (orientation.portrait) {
      navigation.navigate('News');
    }
  };

  return (
    <>
      <TouchableOpacity style={styles.container} onPress={handleClick}>
        <View style={styles.titleContainer}>
          <Text style={styles.title}>{source.name}</Text>
        </View>
        <View style={styles.descriptionContainer}>
          <Text style={styles.description}>{source.description}</Text>
        </View>
        <UrlTextComponent url={source.url} />
      </TouchableOpacity>
      <Divider style={{alignSelf: 'center', width: Mixins.DEVICE_WIDTH90}} />
    </>
  );
};

export default SourceItemComponent;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
  },
  titleContainer: {
    width: '100%',
    justifyContent: 'center',
    marginBottom: 10,
  },
  title: {
    fontSize: Mixins.FONT_SIZE_REGULAR,
    fontWeight: 'bold',
    color: Colors.PRIMARY,
  },
  descriptionContainer: {
    width: '90%',
    height: 60,
    marginBottom: 10,
  },
  description: {
    fontSize: Mixins.FONT_SIZE_SMALL,
    width: '100%',
    fontWeight: '200',
    textAlign: 'justify',
  },
  linkContainer: {
    width: '90%',
  },
});
