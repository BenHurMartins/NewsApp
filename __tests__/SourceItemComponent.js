import 'react-native';
import React from 'react';
import {render, act} from '@testing-library/react-native';
import SourceItemComponent from '../src/components/SourceItemComponent';
import renderer from 'react-test-renderer';

const source = {
  id: 'ben-news',
  url: 'https',
  name: 'Ben Hur News',
  description: 'Lorem ipsum dolor sit amet',
};

describe('SourceItemComponent component', () => {
  it('SourceItemComponent renders correctly', () => {
    renderer.create(<SourceItemComponent source={source} />);
  });
  it('SourceItemComponent snapshot test', () => {
    const tree = renderer
      .create(<SourceItemComponent source={source} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should find the title', () => {
    const {getByText} = render(<SourceItemComponent source={source} />);
    act(() => {
      expect(getByText('Ben Hur News')).not.toBe(null);
    });
  });
  it('Should find the description', () => {
    const {getByText} = render(<SourceItemComponent source={source} />);
    act(() => {
      expect(getByText('Lorem ipsum dolor sit amet')).not.toBe(null);
    });
  });
  it('Should find the url', () => {
    const {getByText} = render(<SourceItemComponent source={source} />);
    act(() => {
      expect(getByText('https')).not.toBe(null);
    });
  });
});
