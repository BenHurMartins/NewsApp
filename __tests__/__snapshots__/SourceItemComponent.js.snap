// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceItemComponent component SourceItemComponent snapshot test 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#65a1dd",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        Ben Hur News
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 60,
          "marginBottom": 10,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 22.5,
            "fontWeight": "200",
            "textAlign": "justify",
            "width": "100%",
          }
        }
      >
        Lorem ipsum dolor sit amet
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20.5,
            "fontWeight": "200",
            "textAlign": "justify",
            "textDecorationLine": "underline",
          }
        }
      >
        https
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#65a1dd",
          "height": 1,
          "width": 750,
        },
        Object {
          "alignSelf": "center",
          "width": 675,
        },
      ]
    }
  />,
]
`;
